app/src/main/java/com/xmartlabs/doapp/ui/TaskAdapter.java

```diff
@@ -0,0 +1,125 @@
+    notifyDataSetChanged();
+  }
+
+  public TaskAdapter(@NonNull Action1<Task> onTaskTapped) {
+    this.onTaskTapped = onTaskTapped;
+  }
+
+  @MainThread
+  public void addTask(@NonNull Task task) {
+    tasks.add(task);
+    notifyDataSetChanged();
+  }
+
+  @Override
+  public TaskHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+    LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
+    View view = layoutInflater.inflate(R.layout.list_item_task, parent, false);
+    return new TaskHolder(view, onTaskTapped);
+  }
+
+  @Override
+  public void onBindViewHolder(TaskHolder holder, int position) {
+    Task task = tasks.get(position);
+    holder.bind(task);
+  }
+
+  @Override
+  public int getItemCount() {
+    return tasks.size();
+  }
+
+  static class TaskHolder extends RecyclerView.ViewHolder {
+    @BindView(R.id.image_view)
+    ImageView doneImageView;
+    @BindView(R.id.color_view)
+    View colorView;
+    @BindView(R.id.description_task)
+    TextView descriptionTextView;
+    @BindView(R.id.title_task)
+    TextView titleTextView;
+    @BindView(R.id.task_linear_layout)
+    LinearLayout taskLinearView;
+
+    @NonNull
+    private final Action1<Task> onTaskTapped;
+
+    public TaskHolder(View itemView, @NonNull Action1<Task> onTaskTapped) {
+      super(itemView);
+      ButterKnife.bind(this, itemView);
+      this.onTaskTapped = onTaskTapped;
+    }
+
+    public void bind(@NonNull Task task) {
+      titleTextView.setText(task.getTitle());
+      if (task.getDescription().isEmpty()) {
+        descriptionTextView.setVisibility(View.GONE);
+      } else {
+        descriptionTextView.setVisibility(View.VISIBLE);
+        descriptionTextView.setText(task.getDescription());
+      }
+      setDoneState(task.isFinished());
+      taskLinearView.setOnClickListener(v -> {
+        setDoneState(!task.isFinished());
+        onTaskTapped.call(task);
+      });
+    }
+
+    private void setDoneState(boolean done) {
+      if (done) {
+        //noinspection deprecation
+        taskLinearView.setBackgroundColor(itemView.getResources().getColor(R.color.seafoam_blue_two_light));
+        //noinspection deprecation
+        colorView.setBackgroundColor(itemView.getResources().getColor(R.color.seafoam_blue_two));
+        //noinspection deprecation
+        doneImageView.setImageDrawable(itemView.getResources().getDrawable(R.drawable.done));
+        //noinspection deprecation
+        doneImageView.setAlpha(VALUE);
+      } else {
+        //noinspection deprecation
+        taskLinearView.setBackgroundColor(itemView.getResources().getColor(R.color.white));
+        //noinspection deprecation
+        colorView.setBackgroundColor(itemView.getResources().getColor(R.color.white));
+        //noinspection deprecation
+        doneImageView.setImageDrawable(itemView.getResources().getDrawable(R.drawable.undo));
+        //noinspection deprecation
+        doneImageView.setAlpha((int) (0.5 * VALUE));
```

Comment:
0.5 should be in the constant too!, I mean, the constant should be  `ALPHA_VALUE`, or something like that